Services/UserService/Commands/EditUserCommand.cs

```diff
@@ -11,29 +16,78 @@ public class EditUserCommand : IEditUserCommand
     {
         private readonly IValidator<UserInfoRequest> userInfoValidator;
         private readonly IValidator<PasswordChangeRequest> passwordChangeValidator;
-        
+        private readonly IBus busControl;
+
         public EditUserCommand
             ([FromServices] IValidator<UserInfoRequest> userInfoValidator, 
-            [FromServices] IValidator<PasswordChangeRequest> passwordChangeValidator)
+            [FromServices] IValidator<PasswordChangeRequest> passwordChangeValidator,
+            [FromServices] IBus busControl)
         {
+            this.busControl = busControl;
             this.userInfoValidator = userInfoValidator;
             this.passwordChangeValidator = passwordChangeValidator;
         }
         
-        public Task<bool> Execute(EditUserRequest request)
+        public async Task<bool> Execute(EditUserRequest request)
         {
             var passwordRequest = request.PasswordRequest;
             var userInfo = request.UserInfo;
-            
-            if (!(string.IsNullOrEmpty(passwordRequest.OldPassword)
-                    || string.IsNullOrEmpty(passwordRequest.NewPassword)))
+            try
+            {
+                if (!(string.IsNullOrEmpty(passwordRequest.OldPassword)
+                        || string.IsNullOrEmpty(passwordRequest.NewPassword)))
+                {
+                    passwordChangeValidator.ValidateAndThrow(passwordRequest);
+                }
+            } 
+            catch 
             {
-                passwordChangeValidator.ValidateAndThrow(passwordRequest);
+                throw new BadRequestException("Unable to edit");
             }
+
             userInfoValidator.ValidateAndThrow(userInfo);
 
-            // TODO Realise edit user logic
-            throw new NotImplementedException();
+            string oldPasswordHash = ShaHash.GetPasswordHash(passwordRequest.OldPassword);
```

Comment:
I forgot: you cannot make this if you have empty password model

Reply:
Fixed